### PR TITLE
e2e summary report infura limit fix

### PR DESCRIPTION
**Pull request by dimpar**
Refs https://github.com/keep-network/tbtc/issues/703

Adding timeout to each `tbtc.js` call from `e2e-summary.js` to limit number of times we call Infura in order to generate an e2e report.

### CODE REVIEW

File: e2e/common.js [hunk 1]
```diff
@@ -104,14 +106,12 @@ export const sendBitcoinTransaction = async(
 }
 
 export const returnBitcoinToDepositor = async (
-    web3,
```
Comment (lukasz-zimnoch) [inline e2e/common.js:107]: We are still passing this parameter in the client code.

File: e2e/assertions.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import {getTBTCTokenBalance, getBtcBalance} from "./common.js";
+import {getBtcBalance} from "./common.js";
+import Web3 from "web3"
```
[the remark this answers is not in the document]
Comment (dimpar) [reply e2e/assertions.js:2]: When I was working on e2e report for `e2e-test.js` I needed to generate an html with data, that in some cases was in `assertions.js`. I wanted to have all the data in 1 file (e2e-test.js). Not passing `web3` was just part of a refactoring to accomplish that. Besides, it looks like the we use web3 only for static (util) purpose to convert a number to BN, I think it reads a bit better when having less params in a function.

File: e2e/common.js (hunk 1 above)
Comment (dimpar) [reply e2e/common.js:107]: Yup, good catch. Removed

File: e2e/assertions.js [hunk 3]
```diff
@@ -14,13 +15,11 @@ export const assertMintedTbtcAmount = (web3, deposit, expectedTbtcAmount) => {
 }
 
 export const assertTbtcAccountBalance = async (
```
Comment (lukasz-zimnoch) [inline e2e/assertions.js:17]: Still used in `e2e-test.js` in the old way (lines 128 and 153)

**Review by lukasz-zimnoch — COMMENTED** [review]
One more comment. Please double check we aligned all of the places where modified functions from `commons` and `assertions` are used. Also, please make sure the e2e test script passes locally.

File: e2e/assertions.js (hunk 3 above)
Comment (dimpar) [reply e2e/assertions.js:17]: These guys were already removed in the [prev](https://github.com/keep-network/local-setup/pull/39/commits/fb36b5bcdc54971cb1a126c523f4e8941ab8814a) commit. Do you have the latest?

Comment (dimpar) [reply e2e/assertions.js:17]: @lukasz-zimnoch I've fixed the input parameters for `assertTbtcAccountBalance()` in `12cb3b1` commit

**Comment by dimpar** [comment]
I've launched VM, setup environment and then run `e2e-test.js` with all the changes in `common.js`, `assertions.js` and test itself. Deposits and redemptions completed successfully. 


<details>
 <summary>Logs from VM</summary>

Initial TBTC balance for account 0xD6b0a1CA8f0641B97eFeC0F1eD73d72e58b38FA5 is: 999000000000000

Starting deposit number [1]...

Creating new deposit contract with lot size 100000 satoshis...
Looking up new deposit with address 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 backed by keep at address 0xf717816C06C2231d464e36Ab79fc2dFcfde4540F...
Monitoring deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 for transition to ACTIVE.
Waiting for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 keep public key...
Waiting for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 to retrieve public key...
sending transaction; attempt number 1
Found public key for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
	Got deposit address: bcrt1qsktjzmyg2pt78eyjknskams96kag95h3dta6kr ; fund with: 100000 satoshis please.
Now monitoring for deposit transaction...
Sending transaction from bcrt1qqytqc2qvg2wtxjw9cd7hsmhdkkmml5lzqc04tk to bcrt1qsktjzmyg2pt78eyjknskams96kag95h3dta6kr
Connecting to electrum server...
Monitoring Bitcoin for transaction to address bcrt1qsktjzmyg2pt78eyjknskams96kag95h3dta6kr...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Transaction 48779853d85dee2416551099a1189500e092e9c8ff676abb59e7a38c6a63bb05 sent
Deposit transaction sent
Received notification for script hash: [0d6dd0f03a849e3d3f734406f9255399a16694ad5923b0f76810fc272eeda0b3] with status: [02dbaa538c2a3938679e94aed14b4d23f933f88b15fbf7f7e43e4f946bb367d0]
Closing connection to electrum server...
Waiting for 1 confirmations for Bitcoin transaction 48779853d85dee2416551099a1189500e092e9c8ff676abb59e7a38c6a63bb05...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Registered listener for blockchain.headers.subscribe event
Received notification of a new block at height: [1163]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Closing connection to electrum server...
Submitting funding proof to deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 for Bitcoin transaction 48779853d85dee2416551099a1189500e092e9c8ff676abb59e7a38c6a63bb05...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 transitioned to ACTIVE.
Deposit is active, minting...
Approving transfer of deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 TDT to Vending Machine...
Minting TBTC...
Found Transfer event for 999500000000000 TBTC.

Deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 has been created successfully.

Starting deposit number [2]...

Creating new deposit contract with lot size 100000 satoshis...
Looking up new deposit with address 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A backed by keep at address 0x5163789853c5270Ca0AF32Fc9502ddf2C82251D8...
Monitoring deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A for transition to ACTIVE.
Waiting for deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A keep public key...
Waiting for deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A to retrieve public key...
sending transaction; attempt number 1
Found public key for deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A...
Monitoring Bitcoin for transaction to address bcrt1q3lngnv2h3nlurs6sjrw373jp8asu6syc3y5dur...
Connecting to electrum server...
	Got deposit address: bcrt1q3lngnv2h3nlurs6sjrw373jp8asu6syc3y5dur ; fund with: 100000 satoshis please.
Now monitoring for deposit transaction...
Sending transaction from bcrt1qqytqc2qvg2wtxjw9cd7hsmhdkkmml5lzqc04tk to bcrt1q3lngnv2h3nlurs6sjrw373jp8asu6syc3y5dur
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Transaction 25962a9713920329f3d19905d34d0845f35342f3de6d0a394b2d9d2d97b728dc sent
Deposit transaction sent
Received notification for script hash: [c9028d79f0665ed7edfc7b5e16a0d8a6f7599a5b4b6209d33771cf3b057e47a2] with status: [0276fcf6535f19b9837eeafd5f1c4dd1601536465815feda2b955f8c946d663e]
Closing connection to electrum server...
Waiting for 1 confirmations for Bitcoin transaction 25962a9713920329f3d19905d34d0845f35342f3de6d0a394b2d9d2d97b728dc...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Registered listener for blockchain.headers.subscribe event
Received notification of a new block at height: [1166]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Closing connection to electrum server...
Submitting funding proof to deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A for Bitcoin transaction 25962a9713920329f3d19905d34d0845f35342f3de6d0a394b2d9d2d97b728dc...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A transitioned to ACTIVE.
Deposit is active, minting...
Approving transfer of deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A TDT to Vending Machine...
Minting TBTC...
Found Transfer event for 999500000000000 TBTC.

Deposit 0xdfF47f74D4bab8Cc4AcDC6FBcAe1624009F4245A has been created successfully.
TBTC balance for account 0xD6b0a1CA8f0641B97eFeC0F1eD73d72e58b38FA5 after performing deposits should be: 2998000000000000. Checking assertion...

Starting redemption of the first deposit...

Using redeemer address: bcrt1q28mc7ukkakkfdlkp745awxzmnfx85ne5h44crq
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Looking up Deposit contract at address 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Looking up Created event for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Found keep address 0xf717816C06C2231d464e36Ab79fc2dFcfde4540F.
Found existing Bitcoin address for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Looking up UTXO size and transaction fee for redemption transaction...
Approving transfer of 1000000000000000 to the vending machine....
Initiating redemption of deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 from vending machine...
Looking up latest redemption details for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Finding or waiting for transaction signature for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Looking for existing signed redemption transaction on Bitcoin chain for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Broadcasting signed redemption transaction to Bitcoin chain for deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Waiting for 1 confirmations for Bitcoin transaction d58195993e4278bd03db7ee1e3c260dc7440377d8aa2ab0f42077fe54b8e7b20...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Registered listener for blockchain.headers.subscribe event
Received notification of a new block at height: [1170]
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Closing connection to electrum server...
Transaction is sufficiently confirmed; submitting redemption proof to deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...


Redemption outcome: Redeemed deposit 0x103EE4FE144d3cE65C7f174a724Bc574c4e2ce53 with Bitcoin transaction d58195993e4278bd03db7ee1e3c260dc7440377d8aa2ab0f42077fe54b8e7b20.

TBTC balance for account 0xD6b0a1CA8f0641B97eFeC0F1eD73d72e58b38FA5 after performing redemption should be: 1998000000000000. Checking assertion...
BTC balance for redeemer address bcrt1q28mc7ukkakkfdlkp745awxzmnfx85ne5h44crq after performing redemption should be: 99850. Checking assertion...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...

Returning redeemed bitcoins to the depositor...

Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Returning 99850 satoshis to depositor bcrt1qqytqc2qvg2wtxjw9cd7hsmhdkkmml5lzqc04tk
Sending transaction from bcrt1q28mc7ukkakkfdlkp745awxzmnfx85ne5h44crq to bcrt1qqytqc2qvg2wtxjw9cd7hsmhdkkmml5lzqc04tk
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Connecting to electrum server...
You are connected to an ElectrumX 1.14.0 server.
Negotiated version: [ElectrumX 1.14.0,1.4.2]
Closing connection to electrum server...
Transaction 4c42ecebe0f84a5cdcbd94401454fce71abeeb640090b23b6c78391cf029f4bf sent
Test completed successfully
vagrant@ubuntu-xenial:~/local-setup$

</details>

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**